src/erlazure_insights.erl

```diff
@@ -0,0 +1,132 @@
+-module(erlazure_insights).
+
+-export([
+        get_events/4,
+        get_time_offset_as_timestamp/2,
+        get_events_all/4,
+        follow_token/3
+    ]).
+
+-include("erlazure.hrl").
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-export([test_url_construction/0]).
+-endif.
+
+-type datetime() :: {{number(), number(), number},
+    {number(), number(), number()}}.
+
+-type offset() :: {seconds | minutes | hours, number()}.
+
+-spec get_events_all(string(), datetime(), offset(), #azure_config{}) ->
+    {ok, list()} | {{error, term()}, {accumulated, list()}}.
+get_events_all(SubscriptionId, To, Offset = {_Unit, _OffsetSeconds}, Config=#azure_config{}) ->
+    {ok, Values, NextToken} = get_events(SubscriptionId, To, Offset, Config),
+    case NextToken of
+        undefined -> {ok, Values};
+        _ -> follow_token(binary_to_list(NextToken), Values, Config)
+    end.
+
+-spec get_events(string(), datetime(), {seconds | minutes | hours, number()}, #azure_config{}) ->
+    {ok, list(), undefined | binary()}.
+get_events(SubscriptionId, To, Offset = {_Unit, _OffsetSeconds}, Config=#azure_config{}) ->
```

Comment:
get_events is just the initial call, all other calls with a skip_token are called by follow_token, this is why I exported that function